lib/supabase/functions.ex

```diff
@@ -84,14 +117,24 @@ defmodule Supabase.Functions do
       {:ok, response} = Supabase.Functions.invoke(client, "my-function", 
         body: %{data: "value"}, 
         timeout: 30_000)
+
+      # With custom auth token
+      {:ok, response} = Supabase.Functions.invoke(client, "my-function", auth: "custom_token")
   """
   @spec invoke(Client.t(), function :: String.t(), opts) :: Supabase.result(Response.t())
   def invoke(%Client{} = client, name, opts \\ []) when is_binary(name) do
     method = opts[:method] || :post
     custom_headers = opts[:headers] || %{}
     timeout = opts[:timeout] || 15_000
 
-    client
+    # Handle auth token override
+    effective_client =
+      case opts[:auth] do
+        nil -> client
+        auth_token when is_binary(auth_token) -> update_auth(client, auth_token)
+      end
```

Comment:
for only 2 logic branches, i would go with a simpler `if/2`, so we can follow the rest of project standard